### PR TITLE
Implemented a Cyborg Analyzer Augment

### PR DESCRIPTION
**Pull request by Ben10083**
Title

### CODE REVIEW

**Comment by Ben10083** [comment]
!review

File: code/modules/organs/subtypes/augment.dm [hunk 1]
```diff
@@ -157,6 +157,19 @@
 	parent_organ = BP_L_HAND
 	aug_slot = slot_l_hand
 
+/obj/item/organ/internal/augment/tool/cAnalyzer
```
Comment (NonQueueingMatt) [inline code/modules/organs/subtypes/augment.dm:160]:
```suggestion
/obj/item/organ/internal/augment/tool/cyborg_analyzer
```

File: code/modules/organs/subtypes/augment.dm [hunk 2]
```diff
@@ -157,6 +157,19 @@
...
+	name = "retractable cyborg analyzer"
+	icon_state = "robotanalyzer"
+	action_button_name = "Deploy Analyzer"
+	action_button_icon = "augment-tool"
+	organ_tag = BP_AUG_CANALYZER
+	parent_organ = BP_R_HAND
+	augment_type = /obj/item/device/robotanalyzer/augment
+
+/obj/item/organ/internal/augment/tool/cAnalyzer/left
```
Comment (NonQueueingMatt) [inline code/modules/organs/subtypes/augment.dm:169]:
```suggestion
/obj/item/organ/internal/augment/tool/cyborg_analyzer/left
```

File: code/__defines/mobs.dm [hunk 3]
```diff
@@ -145,6 +145,7 @@
 #define BP_AUG_TIMEPIECE    "integrated timepiece"
 #define BP_AUG_TOOL         "retractable combitool"
 #define BP_AUG_PEN          "retractable combipen"
+#define BP_AUG_CANALYZER    "retractable cyborg analyzer"
```
Comment (NonQueueingMatt) [inline code/__defines/mobs.dm:148]:
```suggestion
#define BP_AUG_CYBORG_ANALYZER    "retractable cyborg analyzer"
```

File: code/modules/client/preference_setup/loadout/loadout_augments.dm [hunk 4]
```diff
@@ -81,6 +81,20 @@
 	whitelisted = list(SPECIES_HUMAN, SPECIES_HUMAN_OFFWORLD, SPECIES_TAJARA, SPECIES_TAJARA_ZHAN, SPECIES_TAJARA_MSAI, SPECIES_SKRELL, SPECIES_SKRELL_AXIORI, SPECIES_IPC, SPECIES_IPC_G1, SPECIES_IPC_G2, SPECIES_IPC_XION, SPECIES_IPC_ZENGHU, SPECIES_IPC_BISHOP, SPECIES_IPC_SHELL, SPECIES_VAURCA_WORKER, SPECIES_VAURCA_WARRIOR, SPECIES_VAURCA_BULWARK, SPECIES_VAURCA_BREEDER, SPECIES_UNATHI)
 	cost = 3
 
+/datum/gear/augment/analyzer
+	display_name = "retractable cyborg analyzer"
+	description = "An augment that allows the user to deploy a retractable cyborg analyzer."
+	path = /obj/item/organ/internal/augment/tool/cAnalyzer
```
Comment (NonQueueingMatt) [inline code/modules/client/preference_setup/loadout/loadout_augments.dm:87]:
```suggestion
	path = /obj/item/organ/internal/augment/tool/cyborg_analyzer
```

File: code/modules/client/preference_setup/loadout/loadout_augments.dm [hunk 5]
```diff
@@ -81,6 +81,20 @@
...
+	allowed_roles = list("Machinist")
+	cost = 3
+
+/datum/gear/augment/analyzer/New()
+	..()
+	var/list/augs = list()
+	augs["retractable cyborg analyzer, right hand"] = /obj/item/organ/internal/augment/tool/cAnalyzer
```
Comment (NonQueueingMatt) [inline code/modules/client/preference_setup/loadout/loadout_augments.dm:94]:
```suggestion
	augs["retractable cyborg analyzer, right hand"] = /obj/item/organ/internal/augment/tool/cyborg_analyzer
```

File: code/modules/client/preference_setup/loadout/loadout_augments.dm [hunk 6]
```diff
@@ -81,6 +81,20 @@
...
+	augs["retractable cyborg analyzer, left hand"] = /obj/item/organ/internal/augment/tool/cAnalyzer/left
```
Comment (NonQueueingMatt) [inline code/modules/client/preference_setup/loadout/loadout_augments.dm:95]:
```suggestion
	augs["retractable cyborg analyzer, left hand"] = /obj/item/organ/internal/augment/tool/cyborg_analyzer/left
```

File: code/modules/organs/subtypes/augment.dm [hunk 7]
```diff
@@ -157,6 +157,19 @@
 	parent_organ = BP_L_HAND
 	aug_slot = slot_l_hand
 
+/obj/item/organ/internal/augment/tool/cAnalyzer
+	name = "retractable cyborg analyzer"
+	icon_state = "robotanalyzer"
+	action_button_name = "Deploy Analyzer"
+	action_button_icon = "augment-tool"
+	organ_tag = BP_AUG_CANALYZER
```
Comment (NonQueueingMatt) [inline code/modules/organs/subtypes/augment.dm:165]:
```suggestion
	organ_tag = BP_AUG_CYBORG_ANALYZER
```

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
`cAnalyzer` is a bad path name because it doesn't actually tell someone what it is. It's an analyzer, yes, but that "c" could mean anything.

**Comment by Ben10083** [comment]
Requested changes committed and tested to ensure it still works.

**Review by JohnWildkins — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**